app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,50 @@
+class BookingsController < ApplicationController
+  before_action :set_flat, except: [:index, :new]
+
+  def index
+    @bookings = Booking.all
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    @booking.update(booking_params)
+  end
+
+  def new
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(booking_params)
+    @booking.flat = @flat
+    @booking.user = @user
+    if @booking.save!
+      redirect_to user_path(@user)
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def booking_params
+    params.require(:booking).permit(:start_date, :end_date)
```

Comment:
<img width="666" alt="Screenshot 2020-02-01 at 18 59 42" src="https://user-images.githubusercontent.com/9026562/73597566-16644d00-4525-11ea-9e9e-512a8d0101b6.png">

Please see attached. Seems like there is an issue with the start_date and end_date in the schema?